src/commands/test/lib/wizard/uitest.ts

```diff
@@ -0,0 +1,152 @@
+
+interface BuildFolder {
+  name: string;
+  path: string;
+}
+
+export default class RunUitestWizardTestCommand extends AppCommand {
+  private _args: CommandArgs;
+
+  constructor(args: CommandArgs, interactiveArgs: string[]) {
+    super(args);
+    this._args = args;
+    this._args.args.push(...interactiveArgs);
+  }
+
+  public async run(client: AppCenterClient, portalBaseUrl: string): Promise<CommandResult> {
+    const searchFolder: Promise<BuildFolder[]> = this.scanFolder();
+
+    if (this._args.args.indexOf("--async") < 0) {
+      const mergeXml: boolean = await this.promptMergeXml();
+      if (mergeXml) {
+        this._args.args.push("--merge-nunit-xml");
+      }
+    }
+
+    const foundFolders: BuildFolder[] = await searchFolder;
+    const folder: string = await this.promptFolder(foundFolders);
+    this._args.args.push("--build-dir", folder);
+    out.text("\nRunning command: appcenter test run uitest " + this._args.args.join(" ") + "\n");
+    return new RunUITestsCommand(this._args).run(client, portalBaseUrl);
+  }
+
+  private async promptMergeXml(): Promise<boolean> {
+    const questions: Questions = [
+      {
+        type: "list",
+        name: "merge",
+        message: "Should the xml files be merged in to the <output.xml> file?",
+        choices: [{
+          name: "Yes",
+          value: "true"
+        }, {
+          name: "No",
+          value: "false"
+        }]
+      }
+    ];
+    const answers: any = await prompt.question(questions);
+    return answers.merge === "true" ? true : false;
+  }
+
+  private async scanFolder(): Promise<BuildFolder[]> {
+    const foundFolders: BuildFolder[] = [];
+    await this.scanRecurse(process.cwd(), foundFolders);
+    return foundFolders;
+  }
+
+  private async scanRecurse(dirname: string, folders: BuildFolder[]) {
+    const dirContent = fs.readdirSync(dirname);
+    for (const dir of dirContent) {
+      const fullDir = path.join(dirname, dir);
+
+      if (!fs.lstatSync(fullDir).isDirectory()) {
+        continue;
+      }
+      if (dir === "node_modules") {
+        continue;
+      }
+      const dirContents: string[] = fs.readdirSync(fullDir);
+
+      if (dirContents.length === 0) {
+        continue;
+      } else {
+        const xamarinNunitDll = this.findXamarinNunitDll(fullDir);
+        if (!xamarinNunitDll) {
+          await this.scanRecurse(fullDir, folders);
+        } else {
+          const foundFolder: BuildFolder = {
+            name: path.relative(process.cwd(), fullDir),
+            path: fullDir
+          };
+          if (!folders) {
+            folders = [foundFolder];
+          } else {
+            folders.push(foundFolder);
+          }
+        }
+      }
+    }
+  }
+
+  private findXamarinNunitDll(dirname: string): boolean {
+    const dirContent = fs.readdirSync(dirname);
+    let found = 0;
+    for (const dir of dirContent) {
+      const fullDir = path.join(dirname, dir);
+      if (!fs.lstatSync(fullDir).isDirectory()) {
```

Comment:
I don't think we need to check if this is a directory here (although I appreciate why you are) - if someone is naming their directories `Xamarin.UITest.dll` or `nunit.framework.dll` then they're doing very strange things indeed!

Assuming we can remove the `isDirectory` check - you could simplify this function to something like:

```
return dirContent.indexOf("Xamarin.UITest.dll") > -1 && dirContent.indexOf("nunit.framework.dll") > -1
```